packages/map/src/components/UsaMap/components/UsaMap.State.tsx

```diff
@@ -118,6 +119,77 @@ const UsaMap = () => {
     }
   }, [data, state.general.territoriesAlwaysShow])
 
+  useEffect(() => {
+    const passesContrastCheck = []
+    const updatedPatterns = state.map.patterns.map((patternData, patternIndex) => {
+      let updatedPattern = { ...patternData }
+
+      unitedStates.forEach(geo => {
+        const geoKey = geo.properties.iso
+        if (!geoKey) return
+
+        const legendColors = data[geoKey] ? applyLegendToRow(data[geoKey]) : undefined
+        const geoData = data[geoKey]
+        if (!geoData) return
+
+        const hasMatchingValues = patternData.dataValue === geoData[patternData.dataKey]
+        if (!hasMatchingValues) return
+
+        const currentFill = legendColors[0]
+        const patternColor = patternData.color || getContrastColor('#000', currentFill)
+        const contrastCheck = checkColorContrast(currentFill, patternColor)
+
+        // Share with the developer which color is failing as needed...
+        if (contrastCheck === false) {
+          console.warn(`COVE: pattern contrast check failed on ${geoData?.[state.columns.geo.name]} for ${patternData.dataKey} with:
+            pattern color: ${patternColor}
+            contrast: ${getColorContrast(currentFill, patternColor)}
+          `)
+        }
+
+        passesContrastCheck[patternIndex] = contrastCheck
+
+        updatedPattern = {
+          ...updatedPattern,
+          dataValue: geoData[patternData.dataKey],
+          color: patternData.color,
+          size: patternData.size,
+          pattern: patternData.pattern,
+          contrastCheck
+        }
+      })
+
+      return updatedPattern
+    })
+
+    const editorErrorMessage = passesContrastCheck.includes(false) ? 'One or more patterns do not pass the WCAG 2.1 contrast ratio of 3:1.' : ''
+
+    setState(prevState => {
+      // Check if the new editor error message should show or not and update state.runtime
+      const newRuntime = {
+        ...prevState.runtime,
+        editorErrorMessage
+      }
+
+      // If color contrast changes update state.map section
+      const newMap = {
+        ...prevState.map,
+        patterns: updatedPatterns
+      }
+
+      // Only update the state if there are actual changes
+      if (JSON.stringify(prevState.runtime) !== JSON.stringify(newRuntime) || JSON.stringify(prevState.map.patterns) !== JSON.stringify(newMap.patterns)) {
+        return {
+          ...prevState,
+          runtime: newRuntime,
+          map: newMap
+        }
+      }
+
+      return prevState
+    })
+  }, [state.map.patterns, data, applyLegendToRow, state.runtime])
```

Comment:
Looking at this more, my new question is why is it the child's responsibility to tell the parent it should update its state, when the child actual has no affect on the parent's state. 

I don't a reference to the child state anywhere in this useEfffect. 

It seems like the parent should deal with these updates to map.patterns and runtime before it provides those props to the child.

Reply:
Thanks for chewing this over. I removed the useEffect in favor of handling the state change inside of the handleGeoUpdate function which I think makes more sense now.

Reply:
I like the change, much more declarative way of handling it.